app/routes.py

```diff
@@ -1,2 +1,266 @@
+        elif request.args["sort"] == "id": 
+            tasks = Task.query.order_by(Task.id.asc()).all()
+        else:
+            tasks = Task.query.order_by(Task.title.asc()).all()
+    else:
+        tasks = Task.query.all()
+    
+
+    # return jsonify([
+    #     {
+    #         "id": q.id,
+    #         "title": q.title,
+    #         "description": q.description,
+    #         "is_complete": q.is_complete()
+    #     } for q in tasks
+    # ])
+
+    return jsonify([
+        task.as_json() for task in tasks
+    ])
+
+
+@task_list_bp.route("/tasks/<task_id>", methods=["GET", "PUT", "DELETE"])
+def retrieve_one_task(task_id): 
+    task = Task.query.filter_by(id = task_id).first()
+
+    if task is None: 
+        return make_response("", 404)
+
+    if request.method == "GET":   
+        return jsonify({
+                "task": task.as_json()
+        }) 
+
+
+
+       
+    elif request.method == "PUT": 
+        form_data = request.get_json()
+
+        task.title = form_data["title"]
+        task.description = form_data["description"]
+        task.completed_at = form_data["completed_at"]
+
+
+        db.session.commit()
+        
+        return jsonify({
+            "task": task.as_json()
+        })
+
+    elif request.method == "DELETE":
+        db.session.delete(task)
+        db.session.commit()
+        return jsonify (
+        {
+            "details": (f'Task {task.id} "{task.title}" successfully deleted')
+        })
+                
+
+@task_list_bp.route("/tasks/<task_id>/mark_complete", methods=["PATCH"])
+def mark_complete(task_id): 
+    task = Task.query.filter_by(id = task_id).first()
+    if task is None: 
+        return make_response("", 404)
+
+    access_token = os.environ.get("SLACK_BOT_TOKEN")
+    path = "https://slack.com/api/chat.postMessage"
+    response = requests.post(path, data = {
+        "channel": "task-notifications",
+        "text": f"Someone just completed the task {task.title}",
+    }, headers = {
+        "Authorization": access_token,
+    })
+    
+
+    if task.is_complete(): 
+        task.completed_at = datetime.now() 
+        db.session.commit()
+        return jsonify({
+            "task": task.as_json()
+        })
+    else: 
+        task.completed_at = datetime.now()
+        db.session.commit() 
+        return jsonify({
+            "task": task.as_json()
+        })
+        
+
+@task_list_bp.route("/tasks/<task_id>/mark_incomplete", methods=["PATCH"])
+def mark_incomplete(task_id): 
+    task = Task.query.filter_by(id = task_id).first()
+
+    if task is None: 
+        return make_response("", 404)
+
+
+    if task.is_complete(): 
+        task.completed_at = None
+        db.session.commit()
+
+    return jsonify({
+        "task": task.as_json()
+    })
+    
+
+    
+
+#Routes for Goals
+
+goals_bp = Blueprint("goals", __name__, url_prefix="/goals")
+@goals_bp.route("", methods=["POST"])
+def create_a_goal(): 
```

Comment:
👍

You might want to consider putting all your Blueprint declarations at the top together.  It might be more readable to be able to scan all the blueprints at once.